### ISSUE TITLE
Running Built-in Dynamic Optimization using Optimica language extensions example emits a scripting error

### ISSUE THREAD

**Issue by bilderbuchi** [OpenModelica/OpenModelica#8608]
Following the Built-in Dynamic Optimization using Optimica language extensions [example](https://www.openmodelica.org/doc/OpenModelicaUsersGuide/latest/optimization.html#an-example), upon running the optimization, I get an error in the console
```
[3] 13:36:46 Scripting Error
Error parsing expression: readOperator got unknown operator 1.
```

I have visually compared the results with what is in the guide, and they seem to agree, so I don't know what the impact of that error is, but I thought it good to report. 

This is on Windows 10 with OpenModelica v1.19.0-dev-624-g164b90d287 (64-bit)

Ping @vruge , I assume.

Note: I seem to have a different guide coming with this recent nightly (from today) than what is [online](https://www.openmodelica.org/doc/OpenModelicaUsersGuide/latest/optimization.html) as the whole big first section is missing -- I shall study that, too.

**Comment by vruge** [comment]
Hallo @bilderbuchi,

thank you for reporting your observation!
We have a [test](https://github.com/OpenModelica/OpenModelica/blob/master/testsuite/openmodelica/cruntime/optimization/basic/BR.mos)
for the example. The error does not occur in the test, so it may depend on the environment you are using. Can you share some information here?

>  different guide 

The formulation with annotations is more recommended because is native Modelica. The chapter is now further up in the documentation.

**Comment by bilderbuchi** [comment]
Sure, what information are you looking for?

I've now read through the guide for the annotation-based approach, this looks quite nifty! 
I'm having trouble to see how to apply this to a parameter optimisation (since OMOptim is broken/non-functional I can't use that), I'll go to the discussions for this kind of question. May I ping you there?

**Comment by vruge** [comment]
> Sure, what information are you looking for?

Are using OMEdit or mos-script from commandline? 

> I'm having trouble to see how to apply this to a parameter optimisation

the dynamic optimization not support paraemeter estimation.  only optimize the top level inputs

maybe [OMSens](https://openmodelica.org/doc/OpenModelicaUsersGuide/latest/parametersensitivity.html?highlight=omsens) is intressting for you. but I am not familiar with it.

**Comment by bilderbuchi** [comment]
I am using OMedit.

> the dynamic optimization not support paraemeter estimation. 

Ok, bummer, i had hoped that subcase is covered, too. Thanks for the clarification!

I have already made first contact with OMSens 😁
https://github.com/OpenModelica/OMSens/issues/19
I will investigate more

**Comment by vruge** [comment]
@adeas31 is OMEdit drop support for Optimica extensions.
![image](https://user-images.githubusercontent.com/12233943/155753978-dc5db082-d040-49aa-a119-da0f3ae83299.png)

when I try to add a cost term I get
```
Skripting Warnung
Could not preserve the formatting of the model instead internal pretty-printing algorithm is used.
```

Should I include it in the documentation? I mean OMEdit is for Modelica and not for Optimica I was anyway fascinated that it used to work? Since the Annotion approach is preferable I don't find it bad. 
@casella what do you think?

**Comment by adeas31** [comment]
The error only means that the diff algorithm that is used to preserve text formatting failed and pretty-printing algorithm is used. Optimica extensions still might work.

**Comment by vruge** [comment]
Thank you for the explanation. I was somewhat lost after the store parts of the definition had disappeared.
`optimization nmpcBatchReactor(objective = cost)` -> `optimization nmpcBatchReactor`.

The buttons are grayed out for start. So I would be overwhelmed with start the program :smile:

**Comment by bilderbuchi** [comment]
No, that was not it, this was surely set -- without it, the example does not even open because the `optimization` triggers a syntax error.
